### PR TITLE
Services/api integration

### PR DESCRIPTION
**Pull request by MihlaliMgwebi**
- [x] Converted network request promises to Observables in api.js: the observable retrieves flight details from API.
- [x] Cleaned up any subscriptions using observer.complete()
- [x] Made use of RxJS to pipe observables: 
1. pipe used to chain the map operator onto the observable.
2. flight details are mapped to an object that I manipulate in dom.js

### CODE REVIEW

**Comment by MihlaliMgwebi** [comment]
Please make any branching and folder structure suggestions too when it comes to seperation of concerns regarding my observable

File: api.js [hunk 1]
```diff
@@ -10,38 +10,46 @@ const RequestMethods = {
...
-  })
-    .then((responseJSON) => responseJSON.data.states.slice(0, 20))
-    .catch((error) => console.error(error))
-    .finally(() => hideSpinner());
+  return new Observable((observer) => {
+    axios({
+      method: method,
+      url: url,
+    })
+      .then((responseJSON) =>
```
Comment (CharelMarais) [inline api.js:20]: Is this not still a promise within an observable that should be just an observable?

**Review by CharelMarais — COMMENTED** [review]
Looks good, just have a query about the promise noted in the comments.

File: api.js (hunk 1 above)
Comment (MihlaliMgwebi) [reply api.js:20]: As advised, I have used from Fetch.

File: main.js [hunk 2]
```diff
@@ -2,3 +2,17 @@ import { loadFlights, setMinTimeInput } from "./dom.js";
 setMinTimeInput();
 loadFlights();
 
+// import { getFirst20FlightDetails } from "./api.js";
```
Comment (CharelMarais) [inline main.js:5]: just remove the commented code

File: api.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
-import { Observable, map } from "rxjs";
+import { catchError, map, of, switchMap } from "rxjs";
+import { fromFetch } from "rxjs/fetch";
 // [API Calls](https://openskynetwork.github.io/opensky-api/rest.html#all-state-vectors)
```
Comment (CharelMarais) [inline api.js:3]: Remove unused code comment

**Review by CharelMarais — CHANGES_REQUESTED** [review]
Looks good other that the comments.

File: api.js (hunk 3 above)
Comment (MihlaliMgwebi) [reply api.js:3]: Removed and merging.

**Pull request merged**